chore(runtime): Tidy up component paths

grahamking opened the pull request:
There were 13 different ways of referring to a Namespace, Component or Endpoint, lots of legacy names for things.

These are all gone. Most were unused, the others are replaced by either `endpoint.id()` or `component.service_name()`:

```
Namespace
 etcd_path

Component
 instance_root
 path
 etcd_path

Endpoint
 path
 etcd_root
 etcd_path
 etcd_path_with_lease_id
 unique_path
 etcd_path_object_with_lease_id
 name_with_id
 subject_to

etcd
 EtcdPath
 ```

Part of both GA prep and etcd removal.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **Chores**
  * Removed deprecated etcd path-based infrastructure and consolidated internal endpoint identification to use standardized IDs instead of path-based lookups.
  * Cleaned up unused public APIs and internal utilities related to legacy path construction and etcd integration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

grahamking committed:
chore(runtime): Tidy up component paths

There were 13 different ways of referring to a Namespace, Component or
Endpoint, lots of legacy names for things.

These are all gone. Most were unused, the others are replaced by
either `endpoint.id()` or `component.service_name()`:

```
Namespace
 etcd_path

Component
 instance_root
 path
 etcd_path

Endpoint
 path
 etcd_root
 etcd_path
 etcd_path_with_lease_id
 unique_path
 etcd_path_object_with_lease_id
 name_with_id
 subject_to

etcd
 EtcdPath
 ```

Part of both GA prep and etcd removal.

Signed-off-by: Graham King <grahamk@nvidia.com>

grahamking committed:
Thanks Code Rabbit

It caught a very important string prefix match.

Signed-off-by: Graham King <grahamk@nvidia.com>

nnshah1 commented on lib/runtime/src/component/endpoint.rs:
```diff
@@ -259,8 +257,8 @@ impl EndpointConfigBuilder {
-    endpoint_name: &str,
-    subject: &str,
+    endpoint_id: &EndpointId,
+    instance_id: u64,
```
is this connection id in some places?

grahamking replied:
Yup. Renamed.

nnshah1 commented on lib/runtime/src/component/endpoint.rs:
```diff
@@ -288,10 +286,19 @@ fn build_transport_type(
+
+/// The NATS subject to talk to this instance on.
+/// TODO: Do we need to sanitize the names?
+fn nats_subject(endpoint_id: &EndpointId, instance_id: u64) -> String {
```
question: should this be in a nats specific transport file - and not in endpoint?

grahamking replied:
Yes, I thought that too but didn't get to it. Now moved. Thanks for keeping me honest.

nnshah1 reviewed (approved):
LGTM

I'm surprised the push_router still compiled after all the discovery changes

kthui reviewed (approved):
Looks good to me!

Might need a rebase, since there is a [fix](https://github.com/ai-dynamo/dynamo/commit/ef8cf365a778a967fe49bb2823cd75030325c6ff) to the transformer library on main.

grahamking committed:
fix: Move nats subject creation to nats module.

And rename an `instance_id` to `connection_id` to match where it comes
from.

Thanks Neelay!

Signed-off-by: Graham King <grahamk@nvidia.com>
